content/200-concepts/100-components/01-prisma-schema/06-relations/07-referential-actions.mdx

```diff
@@ -0,0 +1,258 @@
+
+</TopBlock>
+
+## What are referential actions?
+
+Referential actions are constraints that define how a referenced record is handled by the database when you run an [`update`](../../prisma-client/crud#update) or [`delete`](../../prisma-client/crud#delete) query.
+
+Prisma supports the following referential actions:
+
+- [`Cascade`](#cascade)
+- [`Restrict`](#restrict)
+- [`NoAction`](#noaction)
+- [`SetNull`](#setnull)
+- [`SetDefault`](#setdefault)
+
+The following example shows `Post` and `User` models with a [one-to-many](./#one-to-many-relations) relationship. The `Post` model has an `authorId` field which is used as the relation scalar for the `author` field, which itself points to the `User` model. 
+
+It's defined using the [`@relation`](../../../../reference/api-reference/prisma-schema-reference#relation) attribute. Under the hood this is represented as a foreign key constraint on the database table.
+
+```prisma file=schema.prisma
+model Post {
+  id        Int     @id @default(autoincrement())
+  title     String
+  author    User    @relation(fields: [authorId], references: [id])
+  authorId  Int
+}
+
+model User {
+  id    Int         @id @default(autoincrement())
+  posts Post[]
+}
+```
+
+The foreign key ensures that when saving any posts, the value of the `authorId` [scalar field](../data-model#scalar-fields) must be referencing an existing  `id` of the `User` model. 
+
+When a [migration](../../prisma-migrate) is run the following SQL is produced:
+
+```sql
+CREATE TABLE "Post" (
+    "id" INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
+    "title" TEXT NOT NULL,
+    "authorId" INTEGER NOT NULL,
+    FOREIGN KEY ("authorId") REFERENCES "User" ("id") ON DELETE RESTRICT ON UPDATE CASCADE
+);
+
+CREATE TABLE "User" (
+    "id" INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT
+);
+```
+
+And the foreign key relation that was created can be seen here:
+
+```sql
+FOREIGN KEY ("authorId") REFERENCES "User" ("id") ON DELETE RESTRICT ON UPDATE CASCADE
+```
+
+## Types of referential actions
+
+### Cascade
+
+When used with `onDelete`, The `Cascade` action deletes all records that are dependant on the record being deleted. 
+
+When used with `onUpdate`, it updates the relation scalar field if the referenced scalar fields of the dependant record are updated.
+
+#### Example usage
+
+```prisma file=schema.prisma highlight=4;add
+model Post {
+  id        Int     @id @default(autoincrement())
+  title     String
+  author    User    @relation(fields: [authorId], references: [id], onDelete: Cascade, onUpdate: Cascade)
+  authorId  Int
+}
+
+model User {
+  id    Int    @id @default(autoincrement())
+  posts Post[]
+}
+```
+
+<Admonition type="info">
+
+`onUpdate` defaults to `Cascade` when not set. This applies to both **optional** and **required** relations. `onDelete` defaults to `Cascade` on **required** relations.
+
+</Admonition>
```

Comment:
I am still confused why this appears here. It is nice to know, but I do not understand the relevance to the example.